src/raftstore/store/peer.rs

```diff
@@ -493,7 +493,16 @@ impl Peer {
                 MessageType::MsgHeartbeat => metrics.heartbeat += 1,
                 MessageType::MsgHeartbeatResponse => metrics.heartbeat_resp += 1,
                 MessageType::MsgTransferLeader => metrics.transfer_leader += 1,
-                MessageType::MsgTimeoutNow => metrics.timeout_now += 1,
+                MessageType::MsgTimeoutNow => {
```

Comment:
No, for example, There is a region has three peer A, B, C, A is leader. A propose an entry e and then tries to transfer leader to B. So the entry e and MsgTimeout will be probably included in the same Ready. So after setting lease to None, it will be updated immediately by e and e can't be dropped obviously.

You can use `util::Either`. `Left` is a normal lease upper bound, `Right` is lease lower bound.